lib/DelegatedModuleFactoryPlugin.js

```diff
@@ -2,55 +2,59 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var DelegatedModule = require("./DelegatedModule");
+"use strict";
+
+const DelegatedModule = require("./DelegatedModule");
 
 // options.source
 // options.type
 // options.context
 // options.scope
 // options.content
-function DelegatedModuleFactoryPlugin(options) {
-	this.options = options;
-	options.type = options.type || "require";
-	options.extensions = options.extensions || ["", ".js"];
-}
-module.exports = DelegatedModuleFactoryPlugin;
+class DelegatedModuleFactoryPlugin {
+	constructor(options) {
+		this.options = options;
+		options.type = options.type || "require";
+		options.extensions = options.extensions || ["", ".js"];
+	}
 
-DelegatedModuleFactoryPlugin.prototype.apply = function(normalModuleFactory) {
-	var scope = this.options.scope;
-	if(scope) {
-		normalModuleFactory.plugin("factory", function(factory) {
-			return function(data, callback) {
-				var dependency = data.dependencies[0];
-				var request = dependency.request;
-				if(request && request.indexOf(scope + "/") === 0) {
-					var innerRequest = "." + request.substr(scope.length);
-					var resolved;
-					if(innerRequest in this.options.content) {
-						resolved = this.options.content[innerRequest];
-						return callback(null, new DelegatedModule(this.options.source, resolved, this.options.type, innerRequest));
-					}
-					for(var i = 0; i < this.options.extensions.length; i++) {
-						var requestPlusExt = innerRequest + this.options.extensions[i];
-						if(requestPlusExt in this.options.content) {
-							resolved = this.options.content[requestPlusExt];
-							return callback(null, new DelegatedModule(this.options.source, resolved, this.options.type, requestPlusExt));
+	apply(normalModuleFactory) {
+		const scope = this.options.scope;
+		if(scope) {
+			normalModuleFactory.plugin("factory", function(factory) {
+				return function(data, callback) {
```

Comment:
`function () {}.bind(this)` is indistinguishable from `() => {}`, besides the latter not having a `prototype` / not being callable with `null`.